source/base/function_cspline.cc

```diff
@@ -0,0 +1,102 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2016 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE at
+// the top level of the deal.II distribution.
+//
+// ---------------------------------------------------------------------
+
+#include <deal.II/base/point.h>
+#include <deal.II/base/function_cspline.h>
+
+#ifdef DEAL_II_WITH_GSL
+#include <cmath>
+#include <algorithm>
+
+DEAL_II_NAMESPACE_OPEN
+namespace Functions
+{
+
+  template <int dim>
+  CSpline<dim>::CSpline(const std::vector<double> &x_,
+                        const std::vector<double> &y_)
+    :
+    x(x_),
+    y(y_),
+    x_max(x.size()>0? x.back()  : 0.),
+    x_min(x.size()>0? x.front() : 0.)
+  {
+    // check that input vector @p x is provided in ascending order:
+    for (unsigned int i = 0; i < x.size()-1; i++)
+      AssertThrow(x[i]<x[i+1],
+                  ExcCSplineOrder(i,x[i],x[i+1]));
+
+    acc = gsl_interp_accel_alloc ();
+    const unsigned int n = x.size();
+    cspline = gsl_spline_alloc (gsl_interp_cspline, n);
+    // gsl_spline_init returns something but it seems nobody knows what
+    const int ierr = gsl_spline_init (cspline, &x[0], &y[0], n);
```

Comment:
You'll probably get a warning about an unused variable here, if you don't use `ierr`.

By the way, the implementation of the function is here: https://fossies.org/dox/gsl-2.1/spline_8c_source.html#l00070
One could try to follow where the return value comes from, but I haven't had much luck with it. Since it's not documented, I suspect that ignoring it is ok, as would asserting that it is zero.